test/run_test.py

```diff
@@ -349,6 +385,9 @@ def parse_args():
         '--ignore-win-blacklist',
         action='store_true',
         help='always run blacklisted windows tests')
+    parser.add_argument(
+        '--determine-from',
+        help='File of affected source filenames to determine which tests to run.')
```

Comment:
Should determination be exposed as a developer mode they can toggle directly? The workflow would be something like, "I change  a Python file, now I want to run every test that might have been affected by that Python file."